beta/src/pages/learn/thinking-in-react.md

```diff
@@ -1,102 +1,113 @@
 
-To implement a UI in React, you will usually follow the same five steps.
+React में UI को लागू करने के लिए, आप आमतौर पर इन्ही पांच स्टेप्स का पालन करेंगे।
 
-## Step 1: Break the UI into a component hierarchy {/*step-1-break-the-ui-into-a-component-hierarchy*/}
+## Step 1: एक कुम्पोनेंट पदानुक्रम में UI को तोड़ो {/_step-1-break-the-ui-into-a-component-hierarchy_/} {/*step-1-एक-कुम्पोनेंट-पदानुक्रम-में-ui-को-तोड़ो-step-1-break-the-ui-into-a-component-hierarchy*/}
 
-Start by drawing boxes around every component and subcomponent in the mockup and naming them. If you work with a designer, they may have already named these components in their design tool. Check in with them!
+मॉकअप में प्रत्येक कौम्पोनॅन्ट और सबकौम्पोनॅन्ट के चारों ओर बॉक्स बनाकर शुरू करें और उनका नामकरण करें।
+यदि आप किसी डिज़ाइनर के साथ काम करते हैं, तो हो सकता है कि उन्होंने अपने डिज़ाइन टूल में इन कॉम्पोनेन्टस का नाम पहले ही रख लिया हो। उनके साथ चेक करलें!
 
-Depending on your background, you can think about splitting up a design into components in different ways:
+आपकी बैकग्राउंड के आधार पर, आप किसी डिज़ाइन को विभिन्न तरीकों से कॉम्पोनेन्टस में बांटने करने के बारे में सोच सकते हैं:
 
-* **Programming**--use the same techniques for deciding if you should create a new function or object. One such technique is the [single responsibility principle](https://en.wikipedia.org/wiki/Single_responsibility_principle), that is, a component should ideally only do one thing. If it ends up growing, it should be decomposed into smaller subcomponents. 
-* **CSS**--consider what you would make class selectors for. (However, components are a bit less granular.)
-* **Design**--consider how you would organize the design's layers.
+- **Programming**--यदि आपको एक नया फ़ंक्शन या ऑब्जेक्ट बनाना चाहिए तो निर्णय लेने के लिए एक ही तकनीक का उपयोग करें। ऐसी तकनीक है [एकल जिम्मेदारी सिद्धांत](https://en.wikipedia.org/wiki/Single_responsibility_principle), यही है, एक कुम्पोनान्ट आदर्श रूप से केवल एक चीज करनी चाहिए। यदि यह बढ़ रहा है, तो इसे छोटे उप कॉम्पोनेन्टस में विघटित किया जाना चाहिए।
 
-If your JSON is well-structured, you'll often find that it naturally maps to the component structure of your UI. That's because UI and data models often have the same information architecture--that is, the same shape. Separate your UI into components, where each component matches one piece of your data model.
+- **CSS**--विचार करें कि आप किसके लिए क्लास चयनकर्ताओं को बनाते हैं. (हालांकि, कुम्पोनान्ट एस थोड़ा कम बारीक हैं।)
+- **Design**--विचार करें कि आप डिज़ाइन परतों को कैसे व्यवस्थित करेंगे।
 
-There are five components on this screen:
+अगर आपका JSON अच्छी तरह से संरचित है, तो आप अक्सर पाएंगे कि यह स्वाभाविक रूप से आपके UI के कंपोनेंट स्ट्रक्चर के लिए मैप करता है। ऐसा इसलिए है क्योंकि UI और डेटा मॉडल में अक्सर एक ही सूचना वास्तुकला होती है - यानी, एक ही आकार है। अपने UI को घटकों में विभाजित करें, जहां प्रत्येक घटक आपके डेटा मॉडल के एक भाग से मेल खाता हो।
```

Comment:
`वास्तुकला` => `आर्किटेक्चर`

let me know if you have any better alternative for this